packages/core/src/render3/instructions.ts

```diff
@@ -241,18 +243,42 @@ export function createLNode(
 /**
  *
  * @param host Existing node to render into.
- * @param renderer Renderer to use.
  * @param template Template function with the instructions.
  * @param context to pass into the template.
  */
-export function renderTemplate<T>(host: LElement, template: ComponentTemplate<T>, context: T) {
+export function renderTemplate<T>(
+    hostNode: RElement, template: ComponentTemplate<T>, context: T,
+    providedRendererFactory: RendererFactory3, host: LElement | null): LElement {
+  if (host == null) {
+    rendererFactory = providedRendererFactory;
+    host = createLNode(
+        null, LNodeFlags.Element, hostNode,
+        createViewState(-1, providedRendererFactory.createRenderer(null, null)));
+  }
   const hostView = host.data !;
   ngDevMode && assertNotEqual(hostView, null, 'hostView');
-  const oldView = enterView(hostView, host);
+  renderComponentOrTemplate(host, hostView, null, template, context);
+  return host;
+}
+
+export function renderComponentOrTemplate<T>(
+    node: LElement, viewState: ViewState, component: T, template?: ComponentTemplate<T>,
```

Comment:
Isn't `component` same as `context`? Can we merge them?

Reply:
Merged, PR updated and rebased